### PR TITLE
GTK+ upgrade

### PR DESCRIPTION
**Pull request by gabi-250**
This PR contains code necessary to port from Tkinter to GTK+.
Other notable changes:
* The use of event (in the form of notifying an observer of changes) to interact with the interface
* A separate thread updates the progress  bar with data from the mixer (as opposed to using a timer to increment the progress bar independently of the progress of the mixer)
* The duration of the media is shown in the playlist in the `minute:second` format

### CODE REVIEW

File: openmedia/tools/timeformatter.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+def _pad_with_zero(numeric_value):
+    if numeric_value < 10:
+        return "0%d" % (numeric_value)
+    else:
+        return "%d" % (numeric_value)
+    
+
```
Comment (gabi-250) [inline openmedia/tools/timeformatter.py:6]: Extra blank line.

File: openmedia/tools/timeformatter.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+def hms_format(initial_seconds):
+    hours = 0
+    minutes = initial_seconds / 60
+    seconds = initial_seconds % 60
```
Comment (gabi-250) [inline openmedia/tools/timeformatter.py]: These assignments are simple enough to be made on a single line I think.

File: openmedia/gui/frames.py [hunk 3]
```diff
@@ -1,189 +1,179 @@
...
 
-    def _play_pause(self, button):
-        self._update_play_icon(button)
+    def _create_playlist(self):
+        self.model = Gio.ListStore.new(ModelItem)
+        self.playlist = Gtk.ListBox()
+        self.playlist.bind_model(self.model, self._create_list_item, None)
+        self.playlist.connect("row_activated", self._play_item)
+        for track in mixer.track_list:
+            item = ModelItem(track.metadata.track_name, track.duration)
```
Comment (gabi-250) [inline openmedia/gui/frames.py:83]: Shouldn't `name` be a property of `Track`? It seems a bit unnecessary to access `metadata` every time the name of a track is needed.

Comment (rbartlensky) [reply openmedia/gui/frames.py:83]: We could create a few properties which would actually use the `metadata` object that is created.

**Comment by rbartlensky** [comment]
I think you can get to squashing some commits.

**Pull request merged**